cli/src/pcluster/validators/networking_validators.py

```diff
@@ -57,6 +63,43 @@ def _validate(self, subnet_ids: List[str]):
             self._add_failure(str(e), FailureLevel.ERROR)
 
 
+class QueueSubnetsValidator(Validator):
+    """
+    Queue Subnets validator.
+
+    Check that there is no duplicate subnet id in the subnet_ids list.
+    Check that subnets in a queue belong to different AZs (EC2 Fleet requests do not support multiple subnets
+    in the same AZ).
+    """
+
+    def _validate(self, queue_name, subnet_ids: List[str]):
+
+        # Test if there are duplicate IDs in subnet_ids
+        if len(set(subnet_ids)) < len(subnet_ids):
+            self._add_failure(
+                "SubnetIds specified in queue {0} contains duplicate subnet IDs.".format(queue_name),
+                FailureLevel.ERROR,
+            )
+
+        # Test if the subnets are all in different AZs
+        try:
+            subnets = AWSApi.instance().ec2.describe_subnets(subnet_ids=subnet_ids)
```

Comment:
This logic of getting the Subnet AZs can moved into the `ec2.py` file for re-use or extend the existing implementation [get_subnet_avail_zone](https://github.com/EddyMM/aws-parallelcluster/blob/50264bbdbf1c540dbe916db7e007b8c7c0e7b6f1/cli/src/pcluster/aws/ec2.py#L113-L113).

I've seen a similar case in the EFA Validator: [az_list](https://github.com/aws/aws-parallelcluster/pull/4544/files#diff-bd4426f9ee10414432866ef60f1b6a41dab292da908dc94ded28ffee6cc66cb3R1912)

Reply:
I think it would be a better experience if the customer was informed about which subnets are in the same AZ, especially since we have this information within this code block.

For duplicate subnets, it's easy for the user to see the duplicate (since there are few AZs/Subnets that can be included).

Reply:
This was addressed in #4547 , on top of which this branch has been rebased.

> This logic of getting the Subnet AZs can moved into the `ec2.py` file for re-use or extend the existing implementation [get_subnet_avail_zone](https://github.com/EddyMM/aws-parallelcluster/blob/50264bbdbf1c540dbe916db7e007b8c7c0e7b6f1/cli/src/pcluster/aws/ec2.py#L113-L113).
> 
> I've seen a similar case in the EFA Validator: [az_list](https://github.com/aws/aws-parallelcluster/pull/4544/files#diff-bd4426f9ee10414432866ef60f1b6a41dab292da908dc94ded28ffee6cc66cb3R1912)

This has been addressed in #4547, on top of which this branch has been rebased.